cmake/Modules/FindLibOpenSSL.cmake

```diff
@@ -15,6 +15,10 @@
 
 if (NOT OPENSSL_FOUND)
 	include (FindPkgConfig)
+	if (APPLE)
+		# Add default pkg-config path of Homebrew’s OpenSSL version
+		set (ENV{PKG_CONFIG_PATH} "$ENV{PKG_CONFIG_PATH}:/usr/local/opt/openssl/lib/pkgconfig")
```

Comment:
This is problematic and breaks non-default setups where the env is already set to something else.

Seems like FindPkgConfig need to be patched so that MacOSX with homebrew is supported. Or we let the situation as before: set PKG_CONFIG_PATH within travis.

Reply:
> This is problematic and breaks non-default setups where the env is already set to something else.

I do not think so. The current setup uses a user provided `PKG_CONFIG_PATH` variable, before it uses the one specified here. Anyway, I now added a check for `ENV{PKG_CONFIG_PATH}`. The current code will now only set `ENV{PKG_CONFIG_PATH}` if it was not set beforehand.

> Seems like FindPkgConfig need to be patched so that MacOSX with homebrew is supported. 

The library `FindPkgConfig` works fine as far as I can tell. The “problem” here is, that `pkg-config` does not find Homebrew’s OpenSSL version if you do not set `PKG_CONFIG_PATH`:

```sh
brew info openssl
#> openssl: stable 1.0.2o (bottled) [keg-only]
#> …
#> For pkg-config to find openssl you may need to set:
#>   set -gx PKG_CONFIG_PATH "/usr/local/opt/openssl/lib/pkgconfig"

pkg-config openssl --variable=libdir
#> Package openssl was not found in the pkg-config search path.
#> Perhaps you should add the directory containing `openssl.pc'
#> to the PKG_CONFIG_PATH environment variable
#> No package 'openssl' found
export PKG_CONFIG_PATH=/usr/local/opt/openssl/lib/pkgconfig
pkg-config openssl --variable=libdir
#> /usr/local/Cellar/openssl/1.0.2o_2/lib
```

.

> Or we let the situation as before: set PKG_CONFIG_PATH within travis.

That did not work correctly on Travis anymore, and was also one of the reasons why I created this PR.